File: backend/internal/models/pair.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Pair struct {
	ID        uuid.UUID `json:"id" db:"id"`
	User1ID   uuid.UUID `json:"user1_id" db:"user1_id"`
	User2ID   uuid.UUID `json:"user2_id" db:"user2_id"`
	User1     *User     `json:"user1,omitempty"`
	User2     *User     `json:"user2,omitempty"`
	CreatedAt time.Time `json:"created_at" db:"created_at"`
}

type CreatePairRequest struct {
	QRCode string `json:"qr_code" binding:"required"`
}

File: iOS/LoveConnection/LoveConnection/Models/Pair.swift
//
//  Pair.swift
//  LoveConnection
//
//  Created on 2025-01-27.
//

import Foundation

struct Pair: Codable, Identifiable {
    let id: UUID
    let user1Id: UUID
    let user2Id: UUID
    let user1: User?
    let user2: User?
    let createdAt: Date
    
    enum CodingKeys: String, CodingKey {
        case id
        case user1Id = "user1_id"
        case user2Id = "user2_id"
        case user1
        case user2
        case createdAt = "created_at"
    }
    
    func getPartner(for userId: UUID) -> User? {
        if user1Id == userId {
            return user2
        } else if user2Id == userId {
            return user1
        }
        return nil
    }
}

